src/services/jsDoc.ts

```diff
@@ -141,33 +147,39 @@ namespace ts.JsDoc {
     }
 
     function getCommentDisplayParts(tag: JSDocTag, checker?: TypeChecker): SymbolDisplayPart[] | undefined {
-        const { comment } = tag;
-        switch (tag.kind) {
+        const { comment, kind } = tag;
+        switch (kind) {
             case SyntaxKind.JSDocImplementsTag:
                 return withNode((tag as JSDocImplementsTag).class);
             case SyntaxKind.JSDocAugmentsTag:
                 return withNode((tag as JSDocAugmentsTag).class);
             case SyntaxKind.JSDocTemplateTag:
-                return addComment((tag as JSDocTemplateTag).typeParameters.map(tp => tp.getText()).join(", "));
+                return addComment((tag as JSDocTemplateTag).typeParameters.map(tp => tp.getText()).join(", "), typeParameterNamePart);
             case SyntaxKind.JSDocTypeTag:
                 return withNode((tag as JSDocTypeTag).typeExpression);
             case SyntaxKind.JSDocTypedefTag:
             case SyntaxKind.JSDocCallbackTag:
             case SyntaxKind.JSDocPropertyTag:
             case SyntaxKind.JSDocParameterTag:
             case SyntaxKind.JSDocSeeTag:
-                const { name } = tag as JSDocTypedefTag | JSDocPropertyTag | JSDocParameterTag | JSDocSeeTag;
-                return name ? withNode(name) : comment === undefined ? undefined : getDisplayPartsFromComment(comment, checker);
+                const { name } = tag as JSDocCallbackTag | JSDocTypedefTag | JSDocPropertyTag | JSDocParameterTag | JSDocSeeTag;
+                return name
+                    ? withNode(name, syntaxKindDisplayPartFunctionDict[kind])
+                    : comment === undefined
+                    ? undefined
+                    : getDisplayPartsFromComment(comment, checker);
             default:
                 return comment === undefined ? undefined : getDisplayPartsFromComment(comment, checker);
         }
 
-        function withNode(node: Node) {
-            return addComment(node.getText());
+        function withNode(node: Node, constructDisplayPartFunc?: ConstructSpecificDisplayPartFunction) {
+            return addComment(node.getText(), constructDisplayPartFunc);
         }
 
-        function addComment(s: string) {
-            return comment ? [textPart(s), spacePart(), ...getDisplayPartsFromComment(comment, checker)] : [textPart(s)];
+        function addComment(s: string, constructDisplayPartFunc?: ConstructSpecificDisplayPartFunction) {
+            return comment
+                ? [!!constructDisplayPartFunc ? constructDisplayPartFunc(s) : textPart(s), spacePart(), ...getDisplayPartsFromComment(comment, checker)]
+                : [textPart(s)];

```

Comment:
this is more a note to myself, but all of jsdoc.ts' displayparts code might need to be reworked to use a DisplayPartWriter in a subsequent PR. It looks like that's what the rest of service uses instead of building display parts themselves.